staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher_whitebox_test.go

```diff
@@ -998,6 +998,15 @@ func TestBookmarksOnResourceVersionUpdates(t *testing.T) {
 	wg.Wait()
 }
 
+type fakeTimeBudget struct {}
+
+func (f *fakeTimeBudget) takeAvailable() time.Duration {
+	return 500*time.Millisecond
```

Comment:
does something actually have to complete within this 500ms or is this just inserting a delay?

Reply:
it's just giving slack - this whole test should be taking less than a second so it's generally about ensuring that we the channel will never be full when we are trying to insert to it (we wait up to these 500ms for inserting)

Reply:
ok... 500ms still seems like something surprisingly high CPU contention could cause issues with. is the issue that the blocked watcher will use up whatever timeout we give it, so we don't want to give it a really long one like wait.ForeverTestTimeout because it would make the test take forever?

Reply:
Yes - this test will consume that period fully exactly once, so that's why I don't want to set very high.
I bumped to 2 seconds - let me know if that's ok for you.